action/action-util/src/main/java/com/powsybl/action/util/PhaseShifterDeltaTapTask.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.action.util;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.contingency.tasks.ModificationTask;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.PhaseTapChanger;
+import com.powsybl.iidm.network.TwoWindingsTransformer;
+
+import java.util.Objects;
+
+/**
+ * @author Hamou AMROUN <hamou.amroun at rte-france.com>
+ */
+public class PhaseShifterDeltaTapTask implements ModificationTask {
+
+    private final String phaseShifterId;
+    private final int tapDelta;
+
+    public PhaseShifterDeltaTapTask(String phaseShifterId, int tapDelta) {
+        this.phaseShifterId = Objects.requireNonNull(phaseShifterId);
+        this.tapDelta = tapDelta;
+    }
+
+    public int getTapDelta() {
+        return tapDelta;
+    }
+
+    @Override
+    public void modify(Network network, ComputationManager computationManager) {
+        Objects.requireNonNull(network);
+        TwoWindingsTransformer phaseShifter = network.getTwoWindingsTransformer(phaseShifterId);
+        if (phaseShifter == null) {
+            throw new PowsyblException("Transformer '" + phaseShifterId + "' not found");
+        }
+        PhaseTapChanger phaseTapChanger = phaseShifter.getPhaseTapChanger();
+        if (phaseTapChanger == null) {
+            throw new PowsyblException("Transformer '" + phaseShifterId + "' is not a phase shifter");
+        }
+        adjustTapPosition(phaseTapChanger);
+        phaseTapChanger.setRegulating(false);
+        phaseTapChanger.setRegulationMode(PhaseTapChanger.RegulationMode.FIXED_TAP);
+    }
+
+    private void adjustTapPosition(PhaseTapChanger phaseTapChanger) {
+        phaseTapChanger.setTapPosition(Math.min(Math.max(phaseTapChanger.getTapPosition() + tapDelta,
+                phaseTapChanger.getLowTapPosition()), phaseTapChanger.getHighTapPosition()));
```

Comment:
To be discussed: do we prefer this implementation or to throw an exception if the target tap is out of range? Personally I prefer to throw, or at least to have a warning.

Reply:
I agree that powsybl user should have a way to know that an action has not fully applied : maybe we could answer this need in further evolutions by returning a richer model of the result of an action application ? Plain logging may not be enough for a proper use in an application.

However, I think throwing here will be too constraining for the user : 
one of the goals of the "delta" implementation is for the user to be able to specify an action which will have more or less the same effect independently from the underlying network state. Throwing depending on the position of the initial tap compared to the limits removes this "feature", and from our business process point of view, it will often prevent the action to be computed without an easy fix.

In our business process, this case is simply handled by the computation possibly saying to the user that the remedial action has not been efficient (not solved constraints). If instead he receives a "not computed", he will not know if the partial application would have been enough or not.